docs/proposals/karmada-operator/custom-leaf-cert-validity/README.md

```diff
@@ -0,0 +1,77 @@
+---
+title: Support to Configure Leaf Certificate Validity Period in Karmada Operator
+authors:
+- "@jabellard"
+reviewers:
+- "@RainbowMango"
+approvers:
+- "@RainbowMango"
+
+creation-date: 2025-02-28
+
+---
+
+# Support to Configure Leaf Certificate Validity Period in Karmada Operator
+
+## Summary
+
+This proposal aims to extend the Karmada operator with support for configuring the validity period of leaf certificates (e.g., API server certificate). By allowing users to specify the validity period in days, this feature provides flexibility to align with organizational security policies and certificate management practices.
+
+## Motivation
+
+In scenarios where security policies require frequent rotation of certificates, the ability to configure the validity period of leaf certificates is essential. This feature ensures that the Karmada control plane can adhere to these policies by allowing users to specify a custom validity period for leaf certificates.
+
+### Goals
+
+- Allow users to specify the validity period of leaf certificates in days.
+- Ensure that the configuration option is optional and defaults to the current behavior if not specified.
+- Enable operators to align Karmada control plane PKI with organizational security policies.
+
+### Non-Goals
+
+- Change the default behavior of the Karmada operator when no custom validity period is provided.
+
+## Proposal
+
+The proposal introduces a new optional field, `LeafCertValidity`, in the `CustomCertificate` struct, where users can specify the validity period of leaf certificates in days.
+
+### API Changes
+
+```go
+type CustomCertificate struct {
+   // Other, existing fields omitted for brevity
+
+   // LeafCertValidity specifies the validity period of leaf certificates (e.g., API Server certificate) in days.
+   // If not specified, the default validity period of 1 year will be used.
+   // +optional
+   LeafCertValidity *int `json:"leafCertValidity,omitempty"`
+}
+```
+
+### User Stories
+
+#### Story 1
+As a cloud infrastructure architect, I want to configure the validity period of leaf certificates to comply with my organization's security policies.
```

Comment:
Just out of curiosity, what's your security policy regarding the validity and rotation of the certificate? (Don't be too specific, as I'm not sure if this is a sensitive thing.)
I heard from other users that they usually set the validity no more than 3 months and rotate cert periodically.

Reply:
3 months is usually a good default.